mowgli_etl/pipeline/wdc/wdc_transformer.py

```diff
@@ -0,0 +1,134 @@
+import json
+from pathlib import Path
+from typing import Gereator, Set, Dict, Union
+from urllib.parse import quote
+import spacy
+
+from mowgli_etl.model.concept_net_predicates import
+from mowgli_etl.model.kg_edge import KgEdge
+from mowtli_etl.model.mowgli_predicates import
+from mowgli_etl.model.kg_node import KgNode
+from mowgli_etl._transformer import _Transformer
+from mowgli_etl.model.word_net_id import WordNetId
+from mowgli_etl.pipeline.wdc.constants import WDC_DATASOURCE_ID
+
+class WDCTransformer(_Transformer):
+    __BAD_DUPLICATE = "\"brand\":"
+
+    def __clean(self, wdc_jsonl_file_path: Path):
+        new_file_name = wdc_jsonl_file_path[0:-6] + "_clean.jsonl"
+
+        with open(wdc_jsonl_file_path, "r") as wdc_jsonl_file_file, open(new_file_name, "w") as new_file:
+            for line in wdc_jsonl_file_file:
+                if line.count(__BAD_DUPLICATE) > 1:
+                    data_starts = []
+                    val = 0
+                    while val != -1:
+                        val = line.find(__BAD_DUPLICATE, val + 1)
+                        if line[val - 2] == ':' or val == -1:
+                            continue
+                        data_starts.append(val - 1)
+
+                    for i in range(len(data_starts)):
+                        if i < len(data_starts - 1):
+                            new_file.write(line[data_starts[i]:data_starts[i+1]] + '\n')
+                        else:
+                            new_file.write(line[data_starts[i]::])
+        
+        return new_file_name
+
+    def __find_dimensions(desctiption, listing, additional_info):
+        dimensions = []
+
+        if description != None:
+            dimensions = re.findall("\d+(?: \d+)?\s?\w*\sx\s\d+\
+                    (?: \d+)?\s?(?:x\s\d+\s?)?\w*", description)
+
+        if len(dimensions) == 0:
```

Comment:
Nit: You may want to structure this as
if dimensions:
   return dimensions
Then fewer indents.